modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -173,52 +173,60 @@ void onScrollResponse(TimeValue delay, SearchResponse searchResponse) {
             total = min(total, mainRequest.getSize());
         }
         task.setTotal(total);
-        task.countThrottle(delay);
-        threadPool.schedule(delay, ThreadPool.Names.GENERIC, threadPool.getThreadContext().preserveContext(new AbstractRunnable() {
+        AbstractRunnable prepareBulkRequest = (AbstractRunnable) threadPool.getThreadContext().preserveContext(new AbstractRunnable() {
```

Comment:
It was confusing to me that you call this `prepareBulkRequest` but also have a method named `prepareBulkRequest`, perhaps one of them can be renamed?